daffodil-core/src/main/scala/org/apache/daffodil/core/dpath/Expression.scala

```diff
@@ -234,6 +223,15 @@ abstract class Expression extends OOLAGHostImpl() with BasicComponent {
       }
       .get
   }
+
+  /**
+   * Path to a single array element without a [N] predicate qualifying it
+   * or to an optional element.
+   *
+   * That is, the kind of expression used to access information
+   * about an array or optional such as its current count.
+   */
+  lazy val isPathToOneWholeArrayOrOptional: Boolean = false
```

Comment:
Can we remove this from `Expression` so it is only a member of `PathExpression`? Seems odd to ever ask a generic `Expression` if it is a special kind of path. Note that `isPathToOneWholeArray` is only a member of PathExpression, so removing this would make the two functions consistent.

Reply:
Yes, moved this to PathExpression.